framework/test/driver/ServiceRegistryTest.cpp

```diff
@@ -88,6 +89,48 @@ void TestUnregisterFix(BundleContext context)
   US_TEST_CONDITION_REQUIRED(!reference.IsConvertibleTo("IBooService"), "Testing for IsConvertibleTo returning false");
 }
 
+void TestServiceReferenceMemberFunctions(BundleContext context)
+{
+  struct TestServiceA : public ITestServiceA
+  {
+  };
+
+  auto s1 = std::make_shared<TestServiceA>();
+  ServiceProperties props;
+  props["StringKey"] = std::string("A string value");
+  props["Status"] = false;
+
+  ServiceRegistration<ITestServiceA> reg1 = context.RegisterService<ITestServiceA>(s1, props);
+  ServiceReference<ITestServiceA> ref1 = context.GetServiceReference<ITestServiceA>();
+
+  // Test ServiceReference member functions GetPropertyKeys()
+  std::vector<std::string> keys;
+  ref1.GetPropertyKeys(keys);
+  US_TEST_CONDITION(keys.size() > 0, "Test keys size")
```

Comment:
Is it possible to write a stricter test? 
If the point of the test is to make sure that the service properties retrieved are correct, checking that more than 1 property exists doesn't seem sufficient.